File: Domain/AppAction.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class AppAction
    {
		public int Id { get; set; }	
		public int Order { get; set; }					
		public virtual ICollection<AppStatusList> FromStatusList { get; set; } 							
		public int ToStatusId { get; set; }						
		public string Action { get; set; }
		public string ActionType { get; set; }
		public string WhenXml { get; set; }
		public int FlowId { get; set; }
		public bool InitStatus { get; set; }
		public int TableId { get; set; }
		public string ActionXml { get; set; }		
    }
}

File: client-app/src/features/AppAction/AppAction.ts
import { AppStatusList } from "../AppStatusList/AppStatusList";

export interface IAppAction {
	Id: number
	Order: number
	Action: string
	ActionType: string	
	FromStatusList : AppStatusList[];
	ToStatusId: number	
	FlowId: number
	InitStatus: boolean
	TableId: number
	WhenXml: string
	ActionXml: string
}

export class AppAction implements IAppAction {
	Id: number = 0;
	Order: number = 0;
	Action: string = '';
	FromStatusList : AppStatusList[] = [];
	ActionType: string = '';
	ToStatusId: number = 0;
	FlowId: number = 0;	
	InitStatus: boolean = false;
	TableId: number = 0;
	WhenXml: string = '';
	ActionXml: string = '';
  
  constructor(init?: IAppAction) {
    Object.assign(this, init);
  }
}

export class AppExport	{
	Order: number = 0;
	ActionType: string = '';
	InitStatus: string = '';
	FromStatus : string = '';
	Action: string = '';		
	ToStatus: string = '';
	WhenXml: string = '';
	ActionXml: string = '';
  
}
	
